src/java.base/share/classes/sun/security/tools/keytool/Main.java

```diff
@@ -4911,13 +4913,23 @@ private void checkWeakConstraint(String label, String sigAlg, Key key,
                 String eMessage = e.getMessage();
                 if (eMessage.contains("denyAfter constraint check failed") &&
                         e.getReason() == BasicReason.ALGORITHM_CONSTRAINED) {
-                    String separator = "java.security: ";
-                    int sepPos = eMessage.indexOf(separator);
-                    String denyAfterDate = "0000-00-00";
-                    if (sepPos > 0) {
-                        denyAfterDate = eMessage.substring(sepPos + separator.length(),
-                                sepPos + separator.length() + denyAfterDate.length());
+                    String startSeparator = "Constraint date: ";
+                    int startSepPos = eMessage.indexOf(startSeparator);
+                    String endSeparator = "; params date";
+                    int endSepPos = eMessage.indexOf(endSeparator);
+                    String denyAfterDate = eMessage.substring(startSepPos + startSeparator.length(),
```

Comment:
I think you should catch `IndexOutOfBoundException` here, just in case, and bail out with your new error message "Unable.to.parse.denyAfter.string.in.exception.message".

Reply:
Fixed.